build-system/tasks/extension-generator/OWNERS

```diff
@@ -0,0 +1,14 @@
+// For an explanation of the OWNERS rules and syntax, see:
+// https://github.com/ampproject/amp-github-apps/blob/master/owners/OWNERS.example
```

Comment:
For posterity, and since I'm not intimately familiar with this directory, can you explain why each of these groups owns it?

Reply:
Added comments. Removed `infra` since it's redundant per parent directory.